components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/template/APIConfigContext.java

```diff
@@ -20,60 +20,56 @@
 
 import org.apache.velocity.VelocityContext;
 import org.wso2.carbon.apimgt.core.exception.ExceptionCodes;
-
-import java.time.Instant;
-import java.time.LocalDateTime;
-import java.time.ZoneId;
-import java.util.Date;
+import org.wso2.carbon.apimgt.core.models.API;
+import org.wso2.carbon.apimgt.core.models.CompositeAPI;
 
 /**
  * Used to generate API meta info related template
  */
 public class APIConfigContext extends ConfigContext {
 
-    private String apiName;
-    private String context;
-    private String version;
-    private LocalDateTime createdTime;
+    private API api;
     private String packageName;
     private String serviceNamePrefix = "";
-    public APIConfigContext(String apiName, String context, String version, LocalDateTime createdTime,
-                            String packageName) {
-        this.apiName = apiName;
-        this.context = context;
-        this.version = version;
-        this.createdTime = createdTime;
+
+    public APIConfigContext(API api, String packageName) {
+        this.api = api;
         this.packageName = packageName;
     }
 
+    public APIConfigContext(CompositeAPI compositeAPI, String gatewayPackageName) {
+
+        this.api = new API.APIBuilder(compositeAPI.getProvider(), compositeAPI.getName(), compositeAPI.getVersion())
```

Comment:
This is a hack! Copying the values of a Composite API into a Standard API is misleading. 

It is better to only store the fields that are needed by the gateway without having a direct dependency on the entire API model object. This allows us to store the common information in the Composite API and Standard API model objects in the same fields without storing other redundant information. Please dont use hacks for new code!